client/src/components/HtmlEditorField/HtmlEditorField.js

```diff
@@ -0,0 +1,46 @@
+/* global window */
+import React from 'react';
+import Script from 'react-load-script';
+import { Component as TextField } from 'components/TextField/TextField';
+import fieldHolder from 'components/FieldHolder/FieldHolder';
+
+class HtmlEditorField extends TextField {
+  constructor(props) {
+    super(props);
+    this.state = {
+        isReady: false
+    };
+    this.handleReady = this.handleReady.bind(this);
+  }
+
+  getInputProps() {
+    return {
+      ...super.getInputProps(),
+      ...this.props.data.config,
+    };
+  }
+
+  handleReady() {
+    if (!window.TinyMCE && window.tinymce) {
+      window.TinyMCE = window.tinymce;
+    }
+    this.setState({ isReady: true });
+  }
+
+  renderDependencyScript() {
+    return <Script url={this.props.data.editorjs} onLoad={this.handleReady} />;
+  }
+
+  renderRichTextEditor() {
+    const config = JSON.parse(this.props.data.config['data-config']);
```

Comment:
this `data.config['data-config']` is verbosely repetitive, what's the reason for the extra `'data-config'` layer, and could it be moved up?

Reply:
This is the data as it's exported from `HTMLEditorConfig`(.php) - it consists of the config, and the editor reference (c.f. `window.ss.editorWrappers`), both of which are required to load the editor, and are applied directly to the element as data-attributes.

Reply:
As per https://github.com/silverstripe/silverstripe-framework/pull/8359#discussion_r215117940 I've renamed the prop to `data.attributes`